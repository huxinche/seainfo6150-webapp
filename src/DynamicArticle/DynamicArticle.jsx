import React from "react";
import HTMLText from "../HTMLText/HTMLText";
import {Link} from "react-router-dom";

const DynamicArticle = (props) => {
  return (
    <>
      <Link to={'/articlelist'}>Back</Link>
      {props.article ?
        <article>
          <header>
            <h1>{props.article.title}</h1>
            <address>
              by {props.article.author} (
              <a href={props.article.authorEmail}>{props.article.authorEmail}</a>)
              <br/>
            </address>
            <time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
          </header>
          <HTMLText text={props.article.text}/>
        </article> : <h3>no article</h3>}
    </>

  );
};

export default DynamicArticle;
