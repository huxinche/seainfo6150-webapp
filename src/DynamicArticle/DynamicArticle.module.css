.back{
  color: #cc0000;
  font-weight: bold;
  cursor: pointer;
  font-size: 18px;
}
.article{
  border-radius: 10px;
  border: 1px dashed gray;
  margin-top: 16px;
  box-shadow: 0 0 14px rgba(200,0,0,0.6);
  padding: 16px 20px;
}
