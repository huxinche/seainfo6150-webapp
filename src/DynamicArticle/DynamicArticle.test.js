import React from "react";
import DynamicArticle from "./DynamicArticle";
import article from '../testData.json'

describe("DynamicArticle tests", () => {
  it("renders correctly", () => {
    const { container } = render(<DynamicArticle article={article} />);
    expect(container).toMatchSnapshot();
  });
});
