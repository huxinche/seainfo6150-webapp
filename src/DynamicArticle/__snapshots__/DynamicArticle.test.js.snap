// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicArticle tests renders correctly 1`] = `
<div>
  <div>
    <a
      class="back"
      href="/articlelist"
    >
      Back
    </a>
    <article
      class="article"
    >
      <header>
        <img
          alt="You can take a pie on a plane"
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
        />
        <h1
          class="title"
        >
          You can take a pie on a plane
        </h1>
        <address>
          by 
          Cathryn Lindsey
           (
          <a
            href="mailto:cathryn.lindsey@nytimes.com"
          >
            cathryn.lindsey@nytimes.com
          </a>
          )
          <br />
        </address>
        <time
          datetime="2018-11-21T22:57:55.000Z"
        >
          November 21st 2018, 2:57 pm
        </time>
      </header>
      <div
        class="content"
      >
        <p>
          Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?
        </p>
        <p>
          First you have to get there.
        </p>
        <p>
          A record 30.6 million passengers will travel on US airlines during the Thanksgiving holiday, up from an estimated 29 million passengers a year ago, according to a forecast by Airlines for America, an US airline trade organization.
        </p>
        <p>
          The forecast translates to an estimated 2.55 million passengers per day, up 137,000 per day from a year ago.
        </p>
        <p>
          The busiest flying day is projected to be Sunday, November 25, with an estimated 3.06 million passengers trying to get home. The second-busiest day will be Wednesday, November 21, with 2.93 million people flying the day before Thanksgiving.
        </p>
        <p>
          The airline association defines the Thanksgiving travel season as starting Friday, November 16, and extending through Tuesday, November 27.
        </p>
        <p>
          The best day to fly during that 12-day period? It's Thanksgiving Day, with just 1.73 million people flying on the actual holiday.
        </p>
        <p>
          The Transportation Security Administration expects to screen about 2.6 million passengers and crew on the Wednesday before the holiday, 500,000 more people than a typical weekday. On November 25, the Sunday after Thanksgiving, the agency estimates more than 2.7 million travelers will be flying home.
        </p>
        <p>
          That's why TSA officials say they're adding more than 1,200 TSA officers and an additional 80 passenger screening canine teams over the holiday, which they define as November 16 through November 26.
        </p>
        <p>
          <q>
            As Thanksgiving and the holiday travel seasons arrive, ensuring the safety and security of the millions of passengers traveling daily remains TSA's top priority,
          </q>
           said TSA Administrator David Pekoske, in a statement.
        </p>
        <p>
          <q>
            We marked several records over the spring and summer travel periods this year, screening more than 525 million passengers and crew. This upcoming season will be very busy as more passengers choose to fly, and TSA officers will be on duty over the holidays so travelers can enjoy theirs.
          </q>
        </p>
      </div>
    </article>
  </div>
</div>
`;
